libsplinter/src/network/mod.rs

```diff
@@ -157,11 +160,41 @@ pub struct Network {
 }
 
 impl Network {
-    pub fn new(mesh: Mesh) -> Self {
-        Network {
+    pub fn new(mesh: Mesh, heartbeat_interval: u64) -> Result<Self, NetworkStartUpError> {
+        let network = Network {
             peers: Arc::new(RwLock::new(PeerMap::new())),
             mesh,
+        };
+
+        if heartbeat_interval != 0 {
+            let heartbeat_network = network.clone();
+            let heartbeat = NetworkHeartbeat::new().write_to_bytes().map_err(|_| {
+                NetworkStartUpError("cannot create NetworkHeartbeat message".to_string())
+            })?;
+            let mut heartbeat_message = NetworkMessage::new();
+            heartbeat_message.set_message_type(NetworkMessageType::NETWORK_HEARTBEAT);
+            heartbeat_message.set_payload(heartbeat);
+            let heartbeat_bytes = heartbeat_message
+                .write_to_bytes()
+                .map_err(|_| NetworkStartUpError("cannot create NetworkMessage".to_string()))?;
+            let _ = thread::spawn(move || {
+                let interval = Duration::from_secs(heartbeat_interval);
+                thread::sleep(interval);
+                loop {
+                    let peers = rwlock_read_unwrap!(heartbeat_network.peers).peer_ids();
```

Comment:
This will ping all peers, temp or otherwise

Reply:
I meant to not turn on the auth guard around the heartbeats so this shouldn't matter, will update

Reply:
fixed